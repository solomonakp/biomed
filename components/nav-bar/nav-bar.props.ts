import { PropsWithChildren } from 'react';

type NavbarBaseProps = PropsWithChildren<object>;

type HTMLNavElementAttributes = React.ComponentPropsWithoutRef<'nav'>;

export type NavBarProps = NavbarBaseProps & HTMLNavElementAttributes;
