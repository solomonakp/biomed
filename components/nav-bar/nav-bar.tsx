import Link from 'next/link';
import { NavBarProps as Props } from './nav-bar.props';

export const NavBar = ({ className = '', ...rest }: Props) => {
    return (
        <nav
            className={`bg-white w-full border-b md:static border-(--color-accent) ${className}`}
            role="navigation"
            aria-label="Navigation bar"
            {...rest}
        >
            <div className="items-center justify-center px-4 max-w-screen-xl mx-auto md:flex md:px-8">
                <div className="flex items-center justify-center py-2 md:py-5 md:block">
                    <header>
                        <Link href="/" className="text-black font-bold text-lg">
                            BioMed
                        </Link>
                    </header>
                </div>
            </div>
        </nav>
    );
};
