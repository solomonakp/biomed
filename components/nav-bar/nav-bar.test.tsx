import { render, screen, cleanup } from '@testing-library/react';
import { axe } from 'jest-axe';

import { NavBar } from './nav-bar';

afterEach(cleanup);

const setUpNavBar = () => render(<NavBar />);

describe('NavBar', () => {
    it('should the Nav bar component be exported correctly', () => {
        expect(NavBar).toBeDefined();
    });
    test('should the Nav bar component be mounted correctly', () => {
        setUpNavBar();

        const navbar = screen.getByRole('navigation');

        expect(navbar).toBeInTheDocument();
    });

    test('should render the Nav bar user would see', () => {
        const { asFragment } = setUpNavBar();

        expect(asFragment()).toMatchSnapshot();
    });
    test('should Nav bar not fail any accessibility test', async () => {
        const { container } = setUpNavBar();

        const result = await axe(container);

        expect(result).toHaveNoViolations();
    });
});
