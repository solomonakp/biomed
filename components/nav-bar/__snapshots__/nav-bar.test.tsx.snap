// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NavBar should render the Nav bar user would see 1`] = `
<DocumentFragment>
  <nav
    aria-label="Navigation bar"
    class="bg-white w-full border-b md:static border-(--color-accent) "
    role="navigation"
  >
    <div
      class="items-center justify-center px-4 max-w-screen-xl mx-auto md:flex md:px-8"
    >
      <div
        class="flex items-center justify-center py-2 md:py-5 md:block"
      >
        <header>
          <a
            class="text-black font-bold text-lg"
            href="/"
          >
            BioMed
          </a>
        </header>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
